pkg/core/resources/apis/meshservice/generate/generator_test.go

```diff
@@ -266,20 +266,26 @@ var _ = Describe("MeshService generator", func() {
 		dp := core_mesh.NewDataplaneResource()
 		Expect(resManager.Delete(context.Background(), dp, store.DeleteByKey("dp-1", model.DefaultMesh))).To(Succeed())
 
+		labelGracePeriodStartedAt := ""
 		// Wait until the MeshService has been marked with grace period start
 		Eventually(func(g Gomega) {
 			g.Expect(resManager.Get(context.Background(), ms, store.GetByKey("backend", model.DefaultMesh))).To(Succeed())
-			g.Expect(ms.GetMeta().GetLabels()).To(HaveKey("kuma.io/deletion-grace-period-started-at"))
+			g.Expect(ms.GetMeta().GetLabels()).To(HaveKey(mesh_proto.DeletionGracePeriodStartedLabel))
+			labelGracePeriodStartedAt = ms.GetMeta().GetLabels()[mesh_proto.DeletionGracePeriodStartedLabel]
 		}, "2s", "100ms").Should(Succeed())
-		labelExistedSince := time.Now()
+
+		gracePeriodStartedAt := time.Now()
+		err = gracePeriodStartedAt.UnmarshalText([]byte(labelGracePeriodStartedAt))
```

Comment:
```suggestion
		gracePeriodStartedAt, err := time.Parse(time.RFC3339, labelGracePeriodStartedAt)
```

Reply:
It's actually invoking `parseStrictRFC3339` inside `UnmarshalText`, and it's the reverse method of the `MarshalText` we are using to generating the label.
See https://github.com/kumahq/kuma/blob/ee63fb3e29ed6e71d7cc730b026836267dc662af/pkg/core/resources/apis/meshservice/generate/generator.go#L230